### PR TITLE
Use a proper table for module TOC

### PR DESCRIPTION
**Pull request by nyalldawson**
And include the brief description of classes in the table.

Before:

image

After:

image

### CODE REVIEW

**Comment by nyalldawson** [comment]
Now groups classes by logical group, eg:

image

image

File: scripts/make_api_rst.py [hunk 1]
```diff
@@ -227,6 +341,43 @@ def generate_docs():
...
+            row_contents = [f"`{class_name} <{class_name}.html>`_", summary or ""]
+            group = ".".join(_class.__group__) if hasattr(_class, "__group__") else ""
+            package_custom_toc[group] += make_table_row(row_contents)
+
+        sorted_package_groups = sorted(
+            package_custom_toc.keys(), key=lambda x: group_names.get(x, x)
+        )
+        if len(sorted_package_groups) > 1:
+            # Add TOC for groups
+            package_index.write(f"\n")
```
Comment (nyalldawson) [inline scripts/make_api_rst.py:357]:
```suggestion
            package_index.write("\n")
```

File: scripts/make_api_rst.py [hunk 2]
```diff
@@ -227,6 +341,43 @@ def generate_docs():
...
+            for package_group in sorted_package_groups:
+                if not package_group:
+                    continue
+
+                anchor = f'{package_name.replace("_", "")}.{package_group}'.replace(".", "_")
+                group_name = group_names.get(package_group, package_group)
+                package_index.write(f"- :ref:`{group_name}<{anchor}>`\n")
+
+            package_index.write(f"\n")
```
Comment (nyalldawson) [inline scripts/make_api_rst.py:366]:
```suggestion
            package_index.write("\n")
```

File: scripts/make_api_rst.py [hunk 3]
```diff
@@ -78,6 +79,73 @@ def ltr_tag(v):
     r"""^(?:([\w.]+::)?([\w.]+\.)?(\w+)\s*(?:\((.*)\)(?:\s*->\s*([\w.]+(?:\[.*?\])?))?(?:\s*\[(signal)\])?)?)?$"""
 )
 
+group_names = {
```
Comment (3nids) [inline scripts/make_api_rst.py:82]: to be coherent, I would move this to https://github.com/qgis/pyqgis-api-docs-builder/blob/main/pyqgis_conf.yml

Comment (nyalldawson) [reply scripts/make_api_rst.py:82]: @3nids done

**Review by 3nids — APPROVED** [review]

**Comment by nyalldawson** [comment]
@3nids what's involved in publishing a new version to the actual site? When's that triggered?

**Comment by 3nids** [comment]
Every push to main on this repo will publish a new version, otherwise it's at least once per week if there is no activity here.

It used the Deb packages, so any change on QGIS source takes a couple of days to be reflected.

**Pull request merged**